Assets/Managers/pipeManager.cs

```diff
@@ -68,13 +69,20 @@ void Update()
 
             //This section deals with transerving out all possible ends of the pipe that is not the entry side
             foreach(Vector2 element in connectingPoints){
-                Debug.Log("STAGE 1");
-                Debug.Log($"checking point {element}");
+                //Debug.Log("STAGE 1");
+                //Debug.Log($"checking point {element}");
                 if(element != connectingPostion && connectingPostion != Vector2.zero){
                     Debug.Log("STAGE 2");
                     Debug.Log($"chosen point {element}");
-                    currentPiece.GetComponent<SpriteRenderer>().color = Color.blue;
-                    tranverse(element,currentPos);
+                    if(waterControl.canMoveDirection(element)){
+                        currentPiece.GetComponent<SpriteRenderer>().color = Color.blue;
+                        tranverse(element,currentPos);
+                    }else{
+                        Debug.Log("Failed state check for direction");
+                        currentPiece.GetComponent<SpriteRenderer>().color = Color.blue;
```

Comment:
Glad we added this, will allow for the "final pipe animation" to go off when the end isn't reached